src/node/internal/internal_dns_client.ts

```diff
@@ -0,0 +1,82 @@
+import * as errorCodes from 'node-internal:internal_dns_constants';
+import { DnsError } from 'node-internal:internal_errors';
+
+export interface Answer {
+  // The record owner.
+  name: string;
+  // The type of DNS record.
+  // These are defined here: https://www.iana.org/assignments/dns-parameters/dns-parameters.xhtml#dns-parameters-4
+  type: number;
+  // The number of seconds the answer can be stored in cache before it is considered stale.
+  TTL: number;
+  // The value of the DNS record for the given name and type. The data will be in text for standardized record types and in hex for unknown types.
+  data: string;
+}
+export interface FailedResponse {
+  error: string;
+}
+export interface SuccessResponse {
+  // The Response Code of the DNS Query.
+  // These are defined here: https://www.iana.org/assignments/dns-parameters/dns-parameters.xhtml#dns-parameters-6
+  Status: number;
+  // If true, it means the truncated bit was set.
+  // This happens when the DNS answer is larger than a single UDP or TCP packet.
+  // TC will almost always be false with Cloudflare DNS over HTTPS because Cloudflare supports the maximum response size.
+  TC: boolean;
+  // If true, it means the Recursive Desired bit was set.
+  RD: boolean;
+  // If true, it means the Recursion Available bit was set.
+  RA: boolean;
+  // 	If true, it means that every record in the answer was verified with DNSSEC.
+  AD: boolean;
+  // 	If true, the client asked to disable DNSSEC validation.
+  CD: boolean;
+  Question: {
+    // The record name requested.
+    name: string;
+    // The type of DNS record requested.
+    // These are defined here: https://www.iana.org/assignments/dns-parameters/dns-parameters.xhtml#dns-parameters-4
+    type: number;
+  }[];
+  Answer?: Answer[];
+  Authority: Answer[];
+  Additional?: Answer;
+}
+
+export async function sendDnsRequest(
+  name: string,
+  type: string
+): Promise<SuccessResponse> {
+  // We are using cloudflare-dns.com and not 1.1.1.1 because of certificate issues.
+  // TODO(soon): Replace this when KJ certificate issues are resolved.
+  const server = new URL('https://cloudflare-dns.com/dns-query');
```

Comment:
Since this will route these requests as public internet requests I guess we should decide if using global fetch is really what we want vs. using a private fetch via a binding that would for sure route the request through internal infrastructure.

Reply:
Yes, for sure, we should find a solution to this before landing this pull-request.